lib/settings/pages/general_settings_page.dart

```diff
@@ -228,7 +229,15 @@ class _GeneralSettingsPageState extends State<GeneralSettingsPage> with SingleTi
       enableCommentNavigation = prefs.getBool(LocalSettings.enableCommentNavigation.name) ?? true;
       combineNavAndFab = prefs.getBool(LocalSettings.combineNavAndFab.name) ?? true;
 
-      openInExternalBrowser = prefs.getBool(LocalSettings.openLinksInExternalBrowser.name) ?? false;
+      browserMode = BrowserMode.values.byName(prefs.getString(LocalSettings.browserMode.name) ?? BrowserMode.customTabs.name);
+      // Migrate the openInExternalBrowser setting, if found.
```

Comment:
I noticed this is done in two different places. Could we maybe set up a system for migrating deprecated preferences?

My initial thought is to have a function that is ran on app startup (e.g., in the `main` function) which performs any required migrations. This can be done in main so that it gets triggered before `ThunderBloc` is initialized!

```dart
// main.dart
performSharedPreferencesMigration();
```

```dart
// lib/utils/preferences/migration.dart

performSharedPreferencesMigration() async {
  // Fetch system preferences
  // Perform migrations here
}
```

Reply:
Good idea! I can see this being used for other things in the future (like a migration of compact mode to an enum). Done!